src/tools/kata-ctl/src/ops/check_ops.rs

```diff
@@ -1,44 +1,198 @@
 // Copyright (c) 2022 Intel Corporation
 //
 // SPDX-License-Identifier: Apache-2.0
-//
+//00
+
+use crate::arch::{x86_64::{check},get_arch_type};
 
-use crate::arch;
-use crate::check;
 use crate::ops::version;
 
 use crate::args::{CheckArgument, CheckSubCommand, IptablesCommand, MetricsCommand};
 
-use anyhow::Result;
+use crate::check::{check_version,run_network_checks};
+
+use anyhow::{anyhow, Result};
+
+use std::fmt::{self,Formatter,Display};
 
 const NAME: &str = "kata-ctl";
 
+#[derive(Debug, Clone, Copy)]
+enum ArchType {
+    x86_64,
+    s390x,
+    aarch64,
+    powerpc64le,
+}
+
+enum UserType {
+    Root,
+    NonRoot,
+}
+
+enum PermissionType {
+    Privileged,
+    NonPrivileged,
+}
+
+type BuiltinCmdFp = fn(args: &str) -> Result<()>;
+
+
+struct CheckItem {
+    name: &'static str,
+    descr: &'static str,
+    fp: BuiltinCmdFp,
+    ut: UserType,
+    arch: ArchType,
+    perm: PermissionType,
+}
+
+static CHECK_LIST: &[CheckItem] = &[
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-network",
+        descr: "This parameter performs network check",
+        fp: run_network_checks,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-version",
+        descr: "Retrieves latest kata version release",
+        fp: check_version,
+        ut: UserType::NonRoot,
+        arch: ArchType::x86_64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::s390x,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::aarch64,
+        perm: PermissionType::NonPrivileged,
+    },
+
+    CheckItem {
+        name: "check-cpu",
+        descr: "This parameter performs the cpu check",
+        fp: check,
+        ut: UserType::NonRoot,
+        arch: ArchType::powerpc64le,
+        perm: PermissionType::NonPrivileged,
+    },
+
+];
+
+impl Display for ArchType {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+
+        match self {
+            ArchType::aarch64 => write!(f, "aarch64"),
+            ArchType::powerpc64le => write!(f, "powerpc64le"),
+            ArchType::s390x => write!(f, "s390x"),
+            ArchType::x86_64 => write!(f, "x86_64"),
+        }
+    }
+}
```

Comment:
This is fine. An alternative is to use the `strum` and `strum-macros` crates to `#[derive(strum_macros::Display)]` on `ArchType` to avoid having to define this function. See https://docs.rs/strum_macros/0.24.3/strum_macros/derive.Display.html.

Reply:
@jodh-intel `strum_macros` used instead of implementing the Display trait.